Towny/src/main/java/com/palmergames/bukkit/towny/command/TownyAdminCommand.java

```diff
@@ -429,8 +433,13 @@ else if (args.length > 2 && TownyCommandAddonAPI.hasCommand(CommandType.TOWNYADM
 							}
 						case "set": {
 							final Town town = TownyUniverse.getInstance().getTown(args[1]);
-							if (town != null)
-								return TownCommand.townSetTabComplete(sender, town, StringMgmt.remArgs(args, 2));
+							if (town != null) {
+								if (args.length == 4) {
+									return NameUtil.filterByStart(adminTownSetTabCompletes, args[3]);
+                                } else {
+									return TownCommand.townSetTabComplete(sender, town, StringMgmt.remArgs(args, 2));
+								}

```

Comment:
Is this actually needed? I think the 
```
if (args.length == 4) {
    blah
}
```
is all that is required for this to work.